pkg/route/hostassignment/externalcertificate.go

```diff
@@ -0,0 +1,34 @@
+package hostassignment
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	routev1 "github.com/openshift/api/route/v1"
+	routecommon "github.com/openshift/library-go/pkg/route"
+)
+
+// ValidateHostExternalCertificate checks if the user has permissions to create and update
+// custom-host subresource of routes. This check is required to be done prior to ValidateHostUpdate()
+// since updating hosts while using externalCertificate is contingent on the user having both these
+// permissions. The ValidateHostUpdate() cannot differentiate if the certificate has changed since
+// now the certificates will be present as a secret object, due to this it proceeds with the assumption
+// that the certificate has changed when the route has externalCertificate set.
```

Comment:
Ultimately, `ValidateUpdate` is going to call `ValidateHostExternalCertificate` and then call `ValidateHostUpdate`, right?  This comment is helpful, but it still isn't quite clear why the logic in `ValidateHostExternalCertificate` cannot simply be added to `ValidateHostUpdate`.

Reply:
Reminded me of [my comment](https://github.com/openshift/library-go/pull/1625#discussion_r1472856895).

Reply:
The EP did mention about invoking `ValidateHostExternalCertificate` prior to `ValidateHostUpdate`, but it's body lacks the actual reason why we decided to choose this approach.

As per our discussion over Slack and further investigation, I've added a `TODO` to consider merging `ValidateHostExternalCertificate` function into `ValidateHostUpdate` later.